test/components/preprocessors/test_markdown_header_splitter.py

```diff
@@ -0,0 +1,324 @@
+    """Test handling of documents with empty content."""
+    splitter_skip = MarkdownHeaderSplitter()  # skip empty documents by default
+    docs = [Document(content="")]
+    result = splitter_skip.run(documents=docs)
+    assert len(result["documents"]) == 0
+
+    splitter_no_skip = MarkdownHeaderSplitter(skip_empty_documents=False)
+    docs = [Document(content="")]
+    result = splitter_no_skip.run(documents=docs)
+    assert len(result["documents"]) == 1
+
+
+def test_split_id_sequentiality_primary_and_secondary(sample_text):
+    # Test primary splitting
+    splitter = MarkdownHeaderSplitter(keep_headers=False)
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Test number of documents
+    assert len(split_docs) == 5
+
+    # Check that split_ids are sequential
+    split_ids = [doc.meta["split_id"] for doc in split_docs]
+    assert split_ids == list(range(len(split_ids)))
+
+    # Test secondary splitting
+    splitter = MarkdownHeaderSplitter(secondary_split="word", split_length=3)
+    docs = [Document(content=sample_text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Test number of documents
+    assert len(split_docs) == 12
+
+    split_ids = [doc.meta["split_id"] for doc in split_docs]
+    assert split_ids == list(range(len(split_ids)))
+
+    # Test with multiple input documents
+    docs = [Document(content=sample_text), Document(content="# Another Header\nSome more content here.")]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+
+    # Test number of documents
+    assert len(split_docs) == 14
+
+    split_ids = [doc.meta["split_id"] for doc in split_docs]
+    assert split_ids == list(range(len(split_ids)))
+
+
+def test_secondary_split_with_overlap():
+    text = (
+        "# Introduction\n"
+        "This is the introduction section with some words for testing overlap splitting. "
+        "It should be split into chunks with overlap.\n"
+        "## Details\n"
+        "Here are more details about the topic. "
+        "Splitting should work across multiple headers and content blocks.\n"
+        "### Subsection\n"
+        "This subsection contains additional information and should also be split with overlap."
+    )
+    splitter = MarkdownHeaderSplitter(secondary_split="word", split_length=4, split_overlap=2, keep_headers=False)
+    docs = [Document(content=text)]
+    result = splitter.run(documents=docs)
+    split_docs = result["documents"]
+    assert len(split_docs) == 21
+
+    for i in range(1, len(split_docs)):
+        prev_doc = split_docs[i - 1]
+        curr_doc = split_docs[i]
+        if prev_doc.meta["header"] == curr_doc.meta["header"]:  # only check overlap within same header
+            prev_words = prev_doc.content.split()
+            curr_words = curr_doc.content.split()
+            assert prev_words[-2:] == curr_words[:2]
```

Comment:
Sorry to be a broken record but let's make this an explicit check for each split. It's easier then for me to review to see if the output is what we expect.